code/datums/components/aura_healing.dm

```diff
@@ -136,9 +145,10 @@
 
 		candidate.updatehealth()
 
-	for (var/mob/remove_alert_from as anything in remove_alerts_from)
-		remove_alert_from.clear_alert(alert_category)
-		current_alerts -= remove_alert_from
+	for (var/datum/weakref/remove_alert_from in remove_alerts_from)
+		var/mob/living/remove_alert_from_real = remove_alert_from.resolve()
+		remove_alert_from_real?.clear_alert(alert_category)
+	current_alerts -= remove_alerts_from
```

Comment:
alsooooo
why remove every entry from current alerts one by one when we can just set it equal to to_heal? (could build it as an assoc list in the loop, that'd make second suggestion possible). I guess it makes insertion more expensive but the lists here aren't that big really, almost doesn't really need to be assoc

also also, rather then filtering remove_alerts_from above, it'll be faster to use assoc lookup to do it here (because you know reasons O(n) vs O(log n), tho I am slowly talking myself out of this idea